ddtrace/tracer/payload.go

```diff
@@ -81,16 +77,22 @@ func (p *payload) size() int {
 	return p.buf.Len() + len(p.header) - p.off
 }
 
-// reset resets the internal buffer, counter and read offset.
+// reset should *not* be used. It is not implemented and is only here to serve
+// as information on how to implement it in case the same payload object ever
+// needs to be reused.
 func (p *payload) reset() {
-	p.off = 8
-	atomic.StoreUint64(&p.count, 0)
-	p.buf.Reset()
-	select {
-	case <-p.closed:
-		// ensure there is room
-	default:
-	}
+	// ⚠️  Warning!
+	//
+	// Resetting the payload for re-use requires the transport to wait for the
+	// HTTP package to Close the request body before attempting to re-use it
+	// again! This requires additional logic to be in place. See:
+	//
+	// • https://github.com/golang/go/blob/go1.16/src/net/http/client.go#L136-L138
+	// • https://github.com/DataDog/dd-trace-go/pull/475
+	// • https://github.com/DataDog/dd-trace-go/pull/549
+	// • https://github.com/DataDog/dd-trace-go/pull/976
+	//
+	panic("not implemented")
```

Comment:
I don't know if this is useful or not. The reasoning was to leave some kind of trail here in order for this to be more easily understandable if someone in the future considers using a pool or re-using the payload. This "wait close" mechanism is hard to miss and if someone ever goes down this path again I figured this would help but could also be overkill. Let me know what you think.

Reply:
I think this is fine. It's good to leave a trail since this issue is so touchy.